x-pack/plugins/maps/public/embeddable/map_embeddable.tsx

```diff
@@ -244,11 +248,22 @@ export class MapEmbeddable extends Embeddable<MapEmbeddableInput, MapEmbeddableO
     return await this._store.dispatch<any>(replaceLayerList(this._layerList));
   }
 
-  addFilters = (filters: Filter[]) => {
-    getUiActions().executeTriggerActions(APPLY_FILTER_TRIGGER, {
-      embeddable: this,
-      filters,
-    });
+  addFilters = (
+    filters: Filter[],
+    { skipTrigger = false }: { skipTrigger?: boolean } = { skipTrigger: false }
+  ) => {
+    if (!skipTrigger) {
+      getUiActions().executeTriggerActions(APPLY_FILTER_TRIGGER, {
+        embeddable: this,
+        filters,
+      });
+    } else {
+      // Applies filters immediately, without intermediate context menu with multiple possible actions
+      getUiActions().getAction(ACTION_APPLY_FILTER).execute({
```

Comment:
@ppisljar, we discussed that we could give this a try and check that it works. 
`ACTION_APPLY_FILTER` is coming from embeddable plugin and not directly dependant on global filter manager. So seems this approach is OK? What do you think?